src/fidesops/schemas/saas/saas_config.py

```diff
@@ -194,6 +211,17 @@ class SaaSConfig(BaseModel):
     test_request: SaaSRequest
     data_protection_request: Optional[SaaSRequest] = None  # GDPR Delete
 
+    @validator("type", pre=True)
+    def validate_log_level(cls, value: str) -> str:
+        """Ensure the provided type is a valid value."""
+        valid_values = [item.value for item in SaaSType]
+        value = value.lower()  # force lowercase, for safety
+
+        if value not in valid_values:
+            raise ValueError(f"Invalid type provided '{value}'.")
+
+        return value
+
```

Comment:
Actually I don't think we need this at all.  `the SaaSType` declaration itself should do this for us, unless you specifically wanted a different message or behavior:
```
{
    "detail": [
        {
            "loc": [
                "body",
                "type"
            ],
            "msg": "value is not a valid enumeration member; permitted: 'mailchimp', 'hubspot', 'outreach', 'segment', 'sentry', 'stripe', 'zendesk', 'custom'",
            "type": "type_error.enum",
            "ctx": {
                "enum_values": [
                    "mailchimp",
                    "hubspot",
                    "outreach",
                    "segment",
                    "sentry",
                    "stripe",
                    "zendesk",
                    "custom"
                ]
            }
        }
    ]
}
```

Reply:
that makes sense